pkg/analysis/optionalorrequired/analyzer.go

```diff
@@ -211,6 +215,33 @@ func reportShouldRemoveSecondaryMarker(field *ast.Field, marker []markers.Marker
 	}
 }
 
+func (a *analyzer) checkTypeSpec(pass *analysis.Pass, typeSpec *ast.TypeSpec, markersAccess markers.Markers) {
+	name := typeSpec.Name.Name
+	set := markersAccess.TypeMarkers(typeSpec)
+
+	for _, marker := range set.UnsortedList() {
+		switch marker.Identifier {
+		case a.primaryOptionalMarker, a.secondaryOptionalMarker, a.primaryRequiredMarker, a.secondaryRequiredMarker:
+			pass.Report(analysis.Diagnostic{
+				Pos:     typeSpec.Pos(),
+				Message: fmt.Sprintf("%s should not be defined on a type %s", marker.String(), name),
```

Comment:
I think this will read more naturally 
```suggestion
				Message: fmt.Sprintf("%s should not be defined on type %s", marker.String(), name),
```